grader/src/main/java/edu/pdx/cs410J/grader/scoring/ProjectSubmissionOutputFileParser.java

```diff
@@ -1,27 +1,106 @@
 package edu.pdx.cs410J.grader.scoring;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
 import java.io.Reader;
+import java.io.StringReader;
+import java.nio.CharBuffer;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeFormatter;
 import java.util.Date;
+import java.util.Locale;
+import java.util.regex.Pattern;
 
 /**
  * Author: Susham Kumar
  */
 
 public class ProjectSubmissionOutputFileParser {
-  public ProjectSubmissionOutputFileParser(Reader source) {
 
+  public ProjectSubmissionOutputFileParser(Reader source) {
+      if(source != null) {
+          bufferedReader = new BufferedReader(source);
+      }
 
   }
 
-  public ProjectSubmission parse() {
-    throw new UnsupportedOperationException("Not implemented yet");
+  public ProjectSubmission parse()  {
+
+     projectSubmission = new ProjectSubmission();
+     String fileContents=null;
+
+      try {
+          while((fileContents= bufferedReader.readLine()) !=null) {
+            fileContents=fileContents.trim();
+          if(fileContents.contains("CS410J")){              // fileContents has the student Id and Project Name.
+              String[] splitContent=fileContents.split(Pattern.quote("."));
+              projectSubmission.setProjectName(splitContent[splitContent.length -1]);
```

Comment:
If the output file being parsed is bogus, we'll end up with `ArrayIndexOutOfBoundsException`s, but I don't know what we could do about it.  Maybe an answer will be become apparent later as we build more of the class..

Reply:
Yes, bogus output file is always going to be an issue.  probably I can write a test case to handle the bogus output file.

As of the first iteration, handling this scenario should help to move on. But, we need to find out a better way or use a different type.

Reply:
Agreed.